viewer/packages/data-providers/src/image-360-data-providers/descriptor-providers/datamodels/cdm/Cdf360CdmBatchCollectionLoader.ts

```diff
@@ -0,0 +1,424 @@
+/*!
+ * Copyright 2025 Cognite AS
+ */
+
+import { CogniteClient, DirectRelationReference, FileInfo } from '@cognite/sdk';
+import { DataModelsSdk } from '../../../../DataModelsSdk';
+import { get360CdmCollectionsQuery, CdfImage360CollectionDmQuery } from './get360CdmCollectionsQuery';
+import type {
+  Historical360ImageSet,
+  Image360Descriptor,
+  Image360FileDescriptor,
+  Image360RevisionId
+} from '../../../../types';
+import type { DMDataSourceType } from '../../../../DataSourceType';
+import chunk from 'lodash/chunk';
+import zip from 'lodash/zip';
+import groupBy from 'lodash/groupBy';
+import partition from 'lodash/partition';
+import { DMInstanceRef } from '@reveal/utilities';
+import { Euler, Matrix4 } from 'three';
+import type { ImageInstanceResult, ImageResultProperties, CollectionInstanceResult, CoreDmFileResponse } from './types';
+import { BATCH_SIZE, BATCH_DELAY_MS } from '../../../../utilities/constants';
+
+type BatchQueryResult = {
+  image_collections: CollectionInstanceResult[];
+  images: ImageInstanceResult[];
+  stations?: Array<{ externalId: string; space: string }>;
+  nextCursor?: Record<string, string>;
+};
+
+interface FileInfoWithInstanceId extends FileInfo {
+  instanceId?: DirectRelationReference;
+}
+
+// Note: DMS properties are returned as string | number | DMInstanceRef (which is DirectRelationReference)
+// We need to handle this flexibility and convert to the correct types when needed
+type DmsPropertyValue = string | number | DMInstanceRef;
```

Comment:
You could use the `RawPropertyValueV3` type from the SDK instead of this, I think 🤔